auto-value-gson-extension/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -869,6 +889,16 @@ private MethodSpec createReadMethod(ClassName className,
       }
     }
 
+    Property unrecognisedJsonPropertiesContainer = properties.stream()
+      .filter(this::isUnrecognisedJsonPropertiesContainer)
+      .findFirst()
+      .orElse(null);
+    if (unrecognisedJsonPropertiesContainer != null) {
+      TypeName mapOfObjects = ParameterizedTypeName.get(LinkedHashMap.class, String.class, Object.class);
```

Comment:
`LinkedHashMap` - why is the order important so that we need to use a linked map?

Reply:
It's convenient for testing when an order of unrecognised fields preserves during deserialisation/serialisation. See `UnrecognisedFieldsTest`.
Do you see any problem with `LinkedHashMap`?

Reply:
[The first answer from stack overflow ](https://stackoverflow.com/a/22839050/3599375) says that it takes 30% more time to create `LinkedHashMap` than `HashMap` **but** iteration over `LinkedHashMap` is 50% faster _(BTW I don't trust house test in terms of access, `LinkedHashMap` can't be faster in access)_. For round trip we need both creation and iteration.